homeassistant/components/pglab/__init__.py

```diff
@@ -0,0 +1,87 @@
+"""PG LAB Electronics integration."""
+from __future__ import annotations
+
+from pypglab.mqtt import Client, Sub_State, Subcribe_CallBack
+
+from homeassistant.components import mqtt
+from homeassistant.components.mqtt import ReceiveMessage
+from homeassistant.components.mqtt.subscription import (
+    async_prepare_subscribe_topics,
+    async_subscribe_topics,
+    async_unsubscribe_topics,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    DEVICE_ALREADY_DISCOVERED,
+    DISCONNECT_COMPONENT,
+    DISCOVERY_INSTANCE,
+    PLATFORMS,
+)
+from .discovery import CreateDiscovery
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up PG LAB Electronics integration from a config entry."""
+
+    # define the call back for pglab  module to publish a mqtt message
+    async def mqtt_publish(
```

Comment:
Why is this an inner function?

Reply:
Tasmota does the same, please see `tasmota/__init__.py` in `async_setup_entry`.

All the 'complexity' of the initial setup that allows pypglab to communicate with mqtt is hidden inside `async_setup_entry`. In my opinion , in this case inner functions make the code more readable.
I can see immediately that `mqtt_publish, mqtt_subscribe, mqtt_unsubscribe` are not used anywhere else, but only important for the pypglab mqtt client setup.

If you really don't like, I can move the inner functions to be global.
Please let me know.

Reply:
As requested I'm moving the callback to be global.
I'm also change the name in mqtt_publish_callback.